.github/workflows/emscripten-patches/embind/bind.cpp

```diff
@@ -0,0 +1,205 @@
+
+template <typename T> static void register_bigint(const char* name) {
+  using namespace internal;
+  _embind_register_bigint(TypeID<T>::get(), name, sizeof(T), std::numeric_limits<T>::min(),
+    std::numeric_limits<T>::max());
+}
+
+template <typename T> static void register_float(const char* name) {
+  using namespace internal;
+  _embind_register_float(TypeID<T>::get(), name, sizeof(T));
+}
+
+// matches typeMapping in embind.js
+enum TypedArrayIndex {
+  Int8Array,
+  Uint8Array,
+  Int16Array,
+  Uint16Array,
+  Int32Array,
+  Uint32Array,
+  Float32Array,
+  Float64Array,
+  // Only available if WASM_BIGINT
+  Int64Array,
+  Uint64Array,
+};
+
+template <typename T> constexpr TypedArrayIndex getTypedArrayIndex() {
+  static_assert(internal::typeSupportsMemoryView<T>(), "type does not map to a typed array");
+  return std::is_floating_point<T>::value
+           ? (sizeof(T) == 4 ? Float32Array : Float64Array)
+           : (sizeof(T) == 1
+                 ? (std::is_signed<T>::value ? Int8Array : Uint8Array)
+                 : (sizeof(T) == 2 ? (std::is_signed<T>::value ? Int16Array : Uint16Array)
+                                   : (sizeof(T) == 4 ? (std::is_signed<T>::value ? Int32Array : Uint32Array)
+                                                     : (std::is_signed<T>::value ? Int64Array : Uint64Array))));
+}
+
+template <typename T> static void register_memory_view(const char* name) {
+  using namespace internal;
+  _embind_register_memory_view(TypeID<memory_view<T>>::get(), getTypedArrayIndex<T>(), name);
+}
+} // namespace
+
+EMSCRIPTEN_BINDINGS(builtin) {
+  using namespace emscripten::internal;
+
+  _embind_register_void(TypeID<void>::get(), "void");
+
+  _embind_register_bool(TypeID<bool>::get(), "bool", sizeof(bool), true, false);
+
+  register_integer<char>("char");
+  register_integer<signed char>("signed char");
+  register_integer<unsigned char>("unsigned char");
+  register_integer<signed short>("short");
+  register_integer<unsigned short>("unsigned short");
+  register_integer<signed int>("int");
+  register_integer<unsigned int>("unsigned int");
+#if __wasm64__
+  register_bigint<signed long>("long");
+  register_bigint<unsigned long>("unsigned long");
+#else
+  register_integer<signed long>("long");
+  register_integer<unsigned long>("unsigned long");
+#endif
+
+  register_bigint<int64_t>("int64_t");
+  register_bigint<uint64_t>("uint64_t");
+
+  register_float<float>("float");
+  register_float<double>("double");
+
+/*cjh
+  _embind_register_std_string(TypeID<std::string>::get(), "std::string");
+  _embind_register_std_string(
+    TypeID<std::basic_string<unsigned char>>::get(), "std::basic_string<unsigned char>");
+  _embind_register_std_wstring(TypeID<std::wstring>::get(), sizeof(wchar_t), "std::wstring");
+  _embind_register_std_wstring(TypeID<std::u16string>::get(), sizeof(char16_t), "std::u16string");
+  _embind_register_std_wstring(TypeID<std::u32string>::get(), sizeof(char32_t), "std::u32string");
```

Comment:
Don't register `std::string` since all code is changed to use `spine::String` which is smaller.

Reply:
Is `bind.cpp` only used for spine?